core/sugarConstructors.mli

```diff
@@ -1,16 +1,13 @@
-module type Pos = sig
-  type t
-  val pos      : t -> Sugartypes.position
-  val with_pos : t -> 'a -> 'a Sugartypes.with_pos
-end
+(* Import module signatures *)
+module type Pos                  = SugarConstructorsIntf.Pos
+module type SugarConstructorsSig = SugarConstructorsIntf.SugarConstructorsSig
 
-module type SmartConstructorsSig = SmartConstructorsIntf.SmartConstructorsSig
+(* Actual implementation of smart constructors as a functor on a Pos module *)
+module SugarConstructors (Position : Pos)
+       : (SugarConstructorsSig with type t := Position.t)
```

Comment:
Comment: You should be able to drop the parentheses in `(SugarConstructorsSig with type t := Position.t)`. Personally, I don't care whether we have a few extraneous parentheses.

Reply:
Yes, these are redundant. I don't know which style is considered better. Personally I found having these parens more readable.